File: src/API/WesternStatesWater.WestDaat.Contracts.Client/Responses/Conservation/ApplicationDashboardLIstItem.cs
using WesternStatesWater.WestDaat.Common.DataContracts;

namespace WesternStatesWater.WestDaat.Contracts.Client.Responses.Conservation;

public class ApplicationDashboardListItem
{
    required public string ApplicantFullName { get; set; }

    required public string ApplicationDisplayId { get; set; }

    required public Guid ApplicationId { get; set; }
    
    required public int CompensationRateDollars { get; set; }

    required public CompensationRateUnits CompensationRateUnits { get; set; }

    required public string OrganizationName { get; set; }

    required public ConservationApplicationStatus Status { get; set; }

    required public DateTimeOffset SubmittedDate { get; set; }

    required public int TotalObligationDollars { get; set; }

    required public double TotalWaterVolumeSavingsAcreFeet { get; set; } 

    required public string WaterRightNativeId { get; set; }
}

File: src/DashboardUI/src/data-contracts/ApplicationDashboardListItem.ts
import { CompensationRateUnits } from './CompensationRateUnits';
import { ConservationApplicationStatus } from './ConservationApplicationStatus';

export interface ApplicationDashboardListItem {
  applicantFullName: string;
  applicationDisplayId: string;
  applicationId: string;
  compensationRateDollars: number;
  compensationRateUnits: CompensationRateUnits;
  organizationName: string;
  status: ConservationApplicationStatus;
  submittedDate: Date;
  totalObligationDollars: number;
  totalWaterVolumeSavingsAcreFeet: number;
  waterRightNativeId: string;
  waterRightState: string;
}

export interface ApplicationDashboardStatistics {
  submittedApplications: number | null;
  approvedApplications: number | null;
  deniedApplications: number | null;
  inReviewApplications: number | null;
  cumulativeEstimatedSavingsAcreFeet: number | null;
  totalObligationDollars: number | null;
}